src/Core/Form/IdentifiableObject/Builder/CartRule/CartRuleActionBuilder.php

```diff
@@ -0,0 +1,141 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Form\IdentifiableObject\Builder\CartRule;
+
+use PrestaShop\PrestaShop\Core\Domain\CartRule\Exception\CartRuleConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\CartRule\ValueObject\CartRuleAction;
+use PrestaShop\PrestaShop\Core\Domain\CartRule\ValueObject\DiscountApplicationType;
+use PrestaShop\PrestaShop\Core\Domain\CartRule\ValueObject\GiftProduct;
+use PrestaShop\PrestaShop\Core\Domain\Currency\ValueObject\CurrencyId;
+use PrestaShop\PrestaShop\Core\Domain\ValueObject\Money;
+use PrestaShop\PrestaShop\Core\Domain\ValueObject\Reduction;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+
+class CartRuleActionBuilder
+{
+    /**
+     * @param array<string, mixed> $actionsData
+     *
+     * @return CartRuleAction
+     */
+    public function build(array $actionsData): CartRuleAction
```

Comment:
I think you can build each condition into a separate method, and then use a public constructor. The goal is to still have an immutable action, and there is no need to edit static methods if we want to add new actions. The builder will then look more nicer :) 

The code will kinda look like this (just an example, needs to be adapted):

```php


class CartRuleActionBuilder {

    public function build(array $data)
    {
        return new CartRuleAction(
            $this->buildFreeShipping(),
            $this->buildGiftProduct(),
            $this->buildAmountDiscount(),
            $this->buildPercentDiscount(),
            $this->buildAmountPercent(),
        );
    }

    private function buildFreeShipping(): bool
    {

    }
    
    private function build(): AmountPercentDiscount
    {

    }
    
    private function buildPercentDiscount(): PercentDiscount
    {
    }
    
    private function buildAmountDiscount(): AmountDiscount
    {

    }
    private function buildGiftProduct(): GiftProduct
    {

    }
}
```